coda/coda_mdstore/views.py

```diff
@@ -826,47 +828,20 @@ def app_bag(request, identifier=None):
 
     if request.method == 'GET' and identifier:
         try:
-            bagObject = Bag.objects.get(name=identifier)
+            bag = Bag.objects.get(name=identifier)
         except Bag.DoesNotExist:
-            return HttpResponseNotFound(
-                "There is no bag with id \'%s\'." % identifier
-            )
-        bagInfoObjectList = Bag_Info.objects.filter(bag_name=identifier)
-        try:
-            url_test = len(re.compile("\w*/\?{2}$").search(
-                request._req.unparsed_uri, 1
-            ).group())
-        except:
-            pass
-        else:
-            if url_test > 2:
-                return HttpResponse(getERC(request, bagObject) + "\n" +\
-                    getERCSupport(request), mimetype='text/plain')
-        # Test url for ERC structure (?)
-        try:
-            url_test = len(re.compile("\w*/\?{1}$").search(
-                request._req.unparsed_uri, 1
-            ).group())
-        except:
-            pass
-        else:
-            if url_test > 1:
-                return HttpResponse(
-                    getERC(request, bagObject), mimetype='text/plain'
-                )
-        returnXML = objectsToXML(bagObject)
-        returnEntry = bagatom.wrapAtom(
-            xml=returnXML,
-            id='http://%s/APP/bag/%s/' % (
-                request.META['HTTP_HOST'], identifier
-            ),
-            title=identifier,
-            author=APP_AUTHOR.get('name', None),
-            author_uri=APP_AUTHOR.get('uri', None)
-        )
-        entryText = XML_HEADER % etree.tostring(returnEntry, pretty_print=True)
-        resp = HttpResponse(entryText, content_type="application/atom+xml")
-        return resp
+            return HttpResponseNotFound("There is no bag with id '{0}'.".format(identifier))
+
+        object_xml = objectsToXML(bag)
+        url = request.build_absolute_uri(reverse('app-bag-detail', args=[identifier]))
```

Comment:
OK, I see what is going on here.

`reverse` is not able to find the `app-bag-detail` because it is missing the namespace defined in `config/urls.py`. The reason that the tests did not find this is becuase we are overriding the main `urls.py` configuration to prevent django from loading *all* of the URLs. 

Originally, this setup was useful because we were testing the apps in isolation as we added tests and upgraded to Django 1.8. I am not sure it is still useful.

I am going to have to take another look at this configuration.